bindings/node.js/kzg.cxx

```diff
@@ -296,73 +274,58 @@ Napi::Value VerifyBlobKzgProof(const Napi::CallbackInfo& info) {
-    Napi::Error::New(env, "verifyBlobKzgProofBatch requires equal number of blobs/commitments/proofs")
-      .ThrowAsJavaScriptException();
-    result = env.Null();
-    goto out;
+  uint32_t count = blobs_param.Length();
+  if (count != commitments_param.Length() || count != proofs_param.Length()) {
+    Napi::Error::New(env, "requires equal number of blobs/commitments/proofs").ThrowAsJavaScriptException();
+    return result;
   }
+  // allocate all three at once. if any fails goto out; will call free on all three so all must exist
+  Blob *blobs = (Blob *)calloc(count, sizeof(Blob));
+  Bytes48 *commitments = (Bytes48 *)calloc(count, sizeof(Bytes48));
+  Bytes48 *proofs = (Bytes48 *)calloc(count, sizeof(Bytes48));
+  if (blobs == nullptr) {
+    Napi::Error::New(env, "Error while allocating memory for blobs").ThrowAsJavaScriptException();
+    goto out;
+  };
+  if (commitments == nullptr) {
+    Napi::Error::New(env, "Error while allocating memory for commitments").ThrowAsJavaScriptException();
+    goto out;
+  };
+  if (proofs == nullptr) {
+    Napi::Error::New(env, "Error while allocating memory for proofs").ThrowAsJavaScriptException();
+    goto out;
+  };
 
-  blobs = (Blob *)calloc(blobs_count, sizeof(Blob));
-   if (blobs == NULL) {
-     Napi::Error::New(env, "Error while allocating memory for blobs").ThrowAsJavaScriptException();
-     result = env.Null();
-     goto out;
-   };
-
-   commitments = (Bytes48 *)calloc(commitments_count, sizeof(Bytes48));
-   if (commitments == NULL) {
-     free(blobs);
-     Napi::Error::New(env, "Error while allocating memory for commitments").ThrowAsJavaScriptException();
-     result = env.Null();
-     goto out;
-   };
-
-   proofs = (Bytes48 *)calloc(proofs_count, sizeof(Bytes48));
-   if (proofs == NULL) {
-     Napi::Error::New(env, "Error while allocating memory for proofs").ThrowAsJavaScriptException();
-     result = env.Null();
-     goto out;
-   };
-
-  for (uint32_t index = 0; index < blobs_count; index++) {
-    // Extract blob bytes from parameter
-    Napi::Value blob = blobs_param[index];
-    auto blob_bytes = blob.As<Napi::Uint8Array>().Data();
-    memcpy(blobs[index].bytes, blob_bytes, BYTES_PER_BLOB);
-
-    // Extract commitment from parameter
-    Napi::Value commitment = commitments_param[index];
-    auto commitment_bytes = commitment.As<Napi::Uint8Array>().Data();
-    memcpy(&commitments[index], commitment_bytes, BYTES_PER_COMMITMENT);
-
-    // Extract proof from parameter
-    Napi::Value proof = proofs_param[index];
-    auto proof_bytes = proof.As<Napi::Uint8Array>().Data();
-    memcpy(&proofs[index], proof_bytes, BYTES_PER_PROOF);
+  for (uint32_t index = 0; index < count; index++) {
+    // add HandleScope here to release reference to temp values
+    // after each iteration since data is being memcpy
+    Napi::HandleScope scope{env};
+    uint8_t *blob = get_bytes(env, blobs_param[index], BYTES_PER_BLOB, "blob");
+    if (blob == nullptr) {
+      goto out;
```

Comment:
Should we be throwing an exception if this happens, like in all the other failure cases (except from bad verification equation)? 

(Same for all the cases above)

Reply:
The JS exception is being thrown in `get_bytes` as (`Napi::Error::New(...).ThrowAsJsException()`) [here](https://github.com/matthewkeil/c-kzg-4844/blob/f10317547860753dddb2633f22121d3412136cc2/bindings/node.js/kzg.cxx#L72). The check for `nullptr` is looking to see if a JS exception was raised to resolve the c stack cleanly.  JS exceptions are handled as application logic and will unwind the application stack, and not an actual c exception (`thow Napi::Error::New(...)` which inherits from `std::exception`) to prevent the c stack from unwinding violently.

Reply:
I added a note about this in the comments that I added to `get_bytes` [here](https://github.com/matthewkeil/c-kzg-4844/blob/f10317547860753dddb2633f22121d3412136cc2/bindings/node.js/kzg.cxx#L52).  Please let me know if I should add some additional comments elsewhere or if you think that paradigm is confusing

Reply:
Makes sense. Thanks!